elevate-cpanel

```diff
@@ -5273,12 +5285,32 @@ EOS
     say <<EOS;
 This is your chance now to cancel the update process.
 You can hit CTRL-C to abort, otherwise the install starts in a few seconds.
+
+It is *highly* recommended that you have a full backup or snapshot
+of your server before proceeding.
 EOS
 
     do { sleep 1; print '.' }
       for 1 .. 6;
     print "\n";
 
+    if ( !$self->getopt('yes') && !$self->getopt('non-interactive') ) {
+        if (
+            !IO::Prompt::prompt(
+                '-one_char',
+                '-yes_no',
+                '-tty',
+                -default => 'n',
+                "Do you wish to proceed with the upgrade process [y/N]: ",
+            )
+        ) {
+            INFO("The user has chosen not to perform the upgrade");
```

Comment:
I'm not sure about the language on this sentence.

Reply:
specifically "the user". Maybe just "Upgrade aborted"?

Reply:
I think we could just exit without INFO here.
For example:
```
# dnf update
-snip-
Total download size: 579 M
Is this ok [y/N]: 
# 
```

Reply:
I was thinking it might be good to have something in the log that showed that the upgrade had been deliberately canceled.  
Maybe something like "The upgrade has been canceled"?

Reply:
Yeah, I can see that "The user has chosen not to perform the upgrade" might seem a more awkward way of saying it.